src/EventProducerService.cpp

```diff
@@ -213,26 +221,54 @@ void EventProducerService::handleProdSvcMessage(const VlcbMessage *msg)
 {
   unsigned int opc = msg->data[0];
   unsigned int nn = Configuration::getTwoBytes(&msg->data[1]);
-  // DEBUG_SERIAL << ">VLCBProdSvc handling message op=" << _HEX(opc) << " nn=" << nn << " en" << en << endl;
-
-  switch (opc) 
-  {    
-
-    case OPC_AREQ:
-      // AREQ message - request for node state, only producer nodes
-
-      if ((nn == module_config->nodeNum) && (eventhandler != nullptr)) 
+  unsigned int en = Configuration::getTwoBytes(&msg->data[3]);
+  
+  if (requesteventhandler != nullptr)
+  {
+    switch (opc)
+    {
+      case OPC_ASRQ:
+        nn = 0000;
+        break;
+        
+      case OPC_AREQ:
+        break;
+        
+      default:
+        return;
+    }
+  
+    byte index = module_config->findExistingEvent(nn, en);
+ 
+    if (index < module_config->EE_MAX_EVENTS)
+    {
+      if (module_config->getEventEVval(index, 1) != 0)
```

Comment:
Please add a comment here to say that we only call the handler for producer events. A producer event is recognised by having EV1 set to an input channel number, i.e. > 0.

Reply:
Done.